Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs

```diff
@@ -21,16 +21,35 @@ public static class ClaimsPrincipalExtensions
         public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
         {
             string userObjectId = claimsPrincipal.GetObjectId();
+            string nameIdentifierId = claimsPrincipal.GetNameIdentifierId();
             string tenantId = claimsPrincipal.GetTenantId();
+            string policyId = claimsPrincipal.GetPolicyId();
 
-            if (!string.IsNullOrWhiteSpace(userObjectId) && !string.IsNullOrWhiteSpace(tenantId))
+            if (!string.IsNullOrWhiteSpace(nameIdentifierId) && !string.IsNullOrWhiteSpace(tenantId) && !string.IsNullOrWhiteSpace(policyId))
             {
+                // B2C standard
```

Comment:
>B2C standard [](start = 19, length = 12)

what is b2c standard?

Reply:
So, this is related to how MSAL generates the `IAccount` identification. When using `AcquireTokenByAuthorizationCode` on B2C apps, the `IAccount` identification will follow the pattern `{oid}-{tfp}.{tid}`, while on AAD apps the pattern is `{oid}.{tid}`. Should I change the word standard to pattern?

Reply:
@TiagoBrenck thanks. that helps. maybe change to pattern and give an example, like you did above?

Reply:
Done